src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,3222 @@
+getTokenomicsInfoV2 blockHash = withUnaryCoreV2 (callV2 @"getTokenomicsInfo") msg (fmap fromProto <$>)
+  where
+    msg = toProto blockHash
+
+getBlocksAtHeightV2 :: (MonadIO m) => BlockHeightInput -> ClientMonad m (GRPCResult [BlockHash])
+getBlocksAtHeightV2 blockHeight = withUnaryCoreV2 (callV2 @"getBlocksAtHeight") msg (fmap fromProto <$>)
+  where
+    msg = toProto blockHeight
+
+getPassiveDelegationInfoV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult QueryTypes.PoolStatus)
+getPassiveDelegationInfoV2 blockHash = withUnaryCoreV2 (callV2 @"getPassiveDelegationInfo") msg (fmap fromProto <$>)
+  where
+    msg = toProto blockHash
+
+getPoolInfoV2 :: (MonadIO m) => BlockHashInput -> BakerId -> ClientMonad m (GRPCResult QueryTypes.PoolStatus)
+getPoolInfoV2 blockHash baker = withUnaryCoreV2 (callV2 @"getPoolInfo") msg (fmap fromProto <$>)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto blockHash & ProtoFields.baker .~ toProto baker
+
+getBlockInfoV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult QueryTypes.BlockInfo)
+getBlockInfoV2 blockHash = withUnaryCoreV2 (callV2 @"getBlockInfo") msg (fmap fromProto <$>)
+  where
+    msg = toProto blockHash
+
+getConsensusInfoV2 :: (MonadIO m) => ClientMonad m (GRPCResult Wasm.WasmModule)
+getConsensusInfoV2 = withUnaryCoreV2 (callV2 @"getModuleSource") defMessage (fmap fromProto <$>)
+
+getModuleSourceV2 :: (MonadIO m) => ModuleRef -> BlockHashInput -> ClientMonad m (GRPCResult Wasm.WasmModule)
+getModuleSourceV2 modRef hash = withUnaryCoreV2 (callV2 @"getModuleSource") msg (fmap fromProto <$>)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto hash & ProtoFields.moduleRef .~ toProto modRef
+
+-- |Retrieve the account information from the chain.
+getAccountInfoV2 ::
+    MonadIO m =>
+    -- | Account identifier, address, index or credential registration id.
+    AccountIdentifierInput ->
+    -- | Block hash
+    BlockHashInput ->
+    ClientMonad m (GRPCResult Concordium.Types.AccountInfo)
+getAccountInfoV2 account blockHash = withUnaryCoreV2 (callV2 @"getAccountInfo") msg (fmap fromProto <$>)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto blockHash & ProtoFields.accountIdentifier .~ toProto account
+
+getInstanceInfoV2 :: (MonadIO m) => ContractAddress -> BlockHashInput -> ClientMonad m (GRPCResult Wasm.InstanceInfo)
+getInstanceInfoV2 cAddress blockHash = withUnaryCoreV2 (callV2 @"getInstanceInfo") msg (fmap fromProto <$>)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto blockHash & ProtoFields.address .~ toProto cAddress
+
+getNextSequenceNumberV2 :: (MonadIO m) => AccountAddress -> ClientMonad m (GRPCResult QueryTypes.NextAccountNonce)
+getNextSequenceNumberV2 accAddress = withUnaryCoreV2 (callV2 @"getNextAccountSequenceNumber") msg (fmap fromProto <$>)
+  where
+    msg = toProto accAddress
+
+{- FIXME: Should we still use GlobalContext for these internally?
```

Comment:
Yes. One is parsed, the protobuf just has byte arrays everywhere.

Reply:
Okay. I ended up adding Rust methods and FFI bindings for instantiating the struct corresponding to the payload in base.